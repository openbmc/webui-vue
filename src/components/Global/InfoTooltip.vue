<template>
  <b-button
    v-b-tooltip
    variant="link"
    class="btn-tooltip btn-icon-only"
    :title="title"
    :aria-label="$t('global.ariaLabel.tooltip')"
  >
    <icon-tooltip />
  </b-button>
</template>

<script>
import IconTooltip from '@carbon/icons-vue/es/information/16';

export default {
  components: { IconTooltip },
  props: {
    title: {
      type: String,
      default: '',
    },
  },
};
</script>

<style lang="scss" scoped>
.btn-tooltip {
  padding: 0;
  line-height: 1em;
  svg {
    vertical-align: baseline;
  }
}
</style>
