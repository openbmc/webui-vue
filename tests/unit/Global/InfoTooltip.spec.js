import { mount, createLocalVue } from '@vue/test-utils';
import InfoTooltip from '@/components/Global/InfoTooltip';
import { BootstrapVue } from 'bootstrap-vue';

process.env.BOOTSTRAP_VUE_NO_WARN = true;
const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('InfoTooltip.vue', () => {
  const wrapper = mount(InfoTooltip, {
    localVue,
    propsData: {
      title: 'A tooltip test title',
    },
    mocks: {
      $t: (key) => key,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
