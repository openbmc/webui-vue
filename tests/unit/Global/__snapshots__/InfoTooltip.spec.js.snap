// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTooltip.vue should render correctly 1`] = `
<template>
  <b-button
    v-b-tooltip
    variant="link"
    class="btn-tooltip"
  >
    <icon-tooltip />
  </b-button>
</template>
`;
